### ISSUE TITLE
SC2320: false negative for printf

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3490]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2320
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
mycommand
# echo "Command exited with $?"       # OK, SC2320 detected
printf 'Command exited with %d\n' $?  # SC2320 NOT detected
if [ $? -ne 0 ]
then
  echo "Failed"
fi
```

#### Here's what shellcheck currently says:
```
Line 5	SC2181: Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.
```

#### Here's what I wanted or expected to see:
```diff
+Line 5	SC2320: This $? refers to echo/printf, not a previous command. Assign to variable to avoid it being overwritten.
 Line 5	SC2181: Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.
```

**Comment by ayushi-work** [comment]
Hi @e-kwsm In `CFG.hs`, `regularExpansionWithStatus` set the exit code token to `getId cmd` where `cmd` was the first command word token (a `T_NormalWord`). However, `checkOverwrittenExitCode` in `Analytics.hs` resolves exit code tokens via `getCommandBasename`, which only works on `T_SimpleCommand` nodes, not bare word tokens. For generic commands, `handleOthers` correctly used the `T_SimpleCommand` ID, so echo worked. For special-cased commands routed through `regularExpansionWithStatus`, the wrong token ID was stored, so the check silently failed.

Fix: Change `regularExpansionWithStatus` to accept the `T_SimpleCommand` ID as an explicit parameter from `handleCommand`, and updated all 10 call sites. Added unit test `prop_checkOverwrittenExitCode3a` for the printf case.
Let me know if this approach is ideal!

**Comment by e-kwsm** [comment]
Hello, @ayushi-work. Thank you for your investigation, but I don&rsquo;t fully understand ShellCheck implementation yet, so I cannot tell if the approach is correct.
PR is welcome, thank you.
